sensor/common/compliance/node_inventory_handler_test.go

```diff
@@ -136,6 +139,62 @@ func (s *NodeInventoryHandlerTestSuite) TestHandlerStopIgnoresError() {
 	s.NoError(h.Stopped().Wait())
 }
 
+// This test simulates a running Sensor loosing connection to Central, followed by a reconnect.
+// As soon as Sensor enters offline mode, it should send NACKs to Compliance.
+// In online mode, inventories are forwarded to Central, which responds with an ACK, that is passed to Compliance.
+func (s *NodeInventoryHandlerTestSuite) TestHandlerOfflineACKNACK() {
+	ch, producer := s.generateTestInputNoClose(1)
+	defer close(ch)
+	h := NewNodeInventoryHandler(ch, &mockAlwaysHitNodeIDMatcher{})
+	h.Notify(common.SensorComponentEventCentralReachable)
+	s.NoError(h.Start())
+	err := mockCentralAck(h)
+	if err != nil {
+		s.Fail(err.Error())
+	}
```

Comment:
```suggestion
	s.NoError(mockCentralAck(h))
```

Reply:
I suggested to get rid of `s` from that function, so maybe the `s.NoError` assertion would need to be handled outside of this function.

Reply:
I've changed `mockCentralAck` to return an `Error` instead of failing the test via `s` directly.
Changing away from `s` here seems a bit counter intuitive to me, as we are in the main test rountine here (`TestHandlerOfflineACKNACK`), so there is not really a place to handle it more outside than here, is it?

Reply:
Yes, but then I would make this func a member of `(s *NodeInventoryHandlerTestSuite)` instead passing it as a parameter.

Reply:
I think, the current state of code looks good, so there is no need to change that more.